core/ormpassword.class.inc.php

```diff
@@ -96,18 +97,18 @@ public function CheckPassword($sClearTextPassword)
 	{
 		$bResult = false;
 		$aInfo = password_get_info($this->m_sHashed);
-		switch ($aInfo["algo"])
+		if (is_null($aInfo["algo"]) || $aInfo["algo"] === 0)
 		{
-			case 0:
-				//unknown, assume it's a legacy password
-				$sHashedPwd = $this->ComputeHash($sClearTextPassword);
-				if ($this->m_sHashed == $sHashedPwd)
-				{
-					$bResult = true;
-				}
-				break;
-			default:
-				$bResult = password_verify($sClearTextPassword, $this->m_sHashed);
+			//unknown, assume it's a legacy password
+			$sHashedPwd = $this->ComputeHash($sClearTextPassword);
+			if ($this->m_sHashed == $sHashedPwd)
+			{
+				$bResult = true;
+			}
```

Comment:
This suggestion is actually a question towards Combodo if this is wanted or not.
```suggestion
			$bResult = ($this->m_sHashed == $sHashedPwd);
```

Reply:
Yep, 💯

Reply:
Okay, @samsk, can you apply the suggestion?